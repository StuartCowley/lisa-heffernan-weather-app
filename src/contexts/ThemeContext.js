import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const ThemeContext = createContext();

export const useThemeContext = () => useContext(ThemeContext);

export const ThemeContextProvider = ({ children }) => {
  const [useDarkTheme, setUseDarkTheme] = useState(false);
  return (
    <ThemeContext.Provider value={{ useDarkTheme, setUseDarkTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeContextProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
