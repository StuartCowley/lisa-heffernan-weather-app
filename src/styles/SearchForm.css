.search-form {
  margin: auto;
  width: fit-content;
}

input {
  margin-left: 2px;
  margin-right:2px;
}

.search-form.light button, .search-form.light input {
  background-color: white;
  border-color: black;
  color: black;
  }

.search-form.dark button, .search-form.dark input {
  background-color: black;
  border-color: white;
  color: white;
} 
