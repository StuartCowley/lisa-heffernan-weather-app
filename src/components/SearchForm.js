import React from "react";
import PropTypes from "prop-types";
import { useThemeContext } from "../contexts/ThemeContext";
import "../styles/SearchForm.css";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const { useDarkTheme } = useThemeContext();
  const handleChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <div className={`search-form ${useDarkTheme ? "dark" : "light"}`}>
      <input className="search-form__input" type="text" onChange={handleChange} value={searchText} />
      <button className="search-form__button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
};

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
