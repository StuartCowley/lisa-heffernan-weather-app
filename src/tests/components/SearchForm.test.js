import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";
import { ThemeContextProvider } from "../../contexts/ThemeContext";

describe("SearchForm", () => {
  const validProps = {
    searchText: "manchester",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ThemeContextProvider>
        <SearchForm
          searchText={validProps.searchText}
          setSearchText={validProps.setSearchText}
          onSubmit={validProps.onSubmit}
        />
      </ThemeContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct text", () => {
    const { getByText } = render(
      <ThemeContextProvider>
        <SearchForm
          searchText={validProps.searchText}
          setSearchText={validProps.setSearchText}
          onSubmit={validProps.onSubmit}
        />
      </ThemeContextProvider>,
    );

    expect(getByText("Search")).toHaveClass("search-form__button");
  });
});
