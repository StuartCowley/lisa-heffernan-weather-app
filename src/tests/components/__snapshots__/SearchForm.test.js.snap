// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form light"
  >
    <input
      class="search-form__input"
      type="text"
      value="manchester"
    />
    <button
      class="search-form__button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
